### DISCUSSION TITLE
Incremental Static Generation not working with CDN path on assetPrefix

### DISCUSSION THREAD

**Discussion by matteoferigo** [vercel/next.js#23979] [Help]
# Bug report

## Describe the bug

Fallback option for `getStaticPaths` not working properly when `assetPrefix` is set in `next.config.js`

## To Reproduce

Set the CDN path in the `assetPrefix`

```javascript
module.exports = {
  assetPrefix: 'https://cdn-name.com'
}
```

Create a page component with a `fallback: true` option in `getStaticPaths`

```javascript
const AwesomePage = ({ data }) => {
  const router = useRouter()

  if (router.isFallback) return <div>Loading...</div>

  return <div>{`My awesome data: ${data}`}</div>
}

export const getStaticPaths = async () => ({
  paths: languages.map((lang) => ({ params: { lang } })),
  fallback: true,
})

export const getStaticProps = async (ctx) => {
  const data = await request(ctx)
  return { props: { ...ctx.params, data } }
}

export default AwesomePage
```

When you load the page you will see that the page is built at each request.
You can see it because of the fallback component displayed anytime you refresh the page.

By removing `assetPrefix` from `next.config.js`, page behavior is correct 

## Expected behavior

It should build the page only the first time and dispatch it for any other request… so you should see the fallback component just once.

An approach should be something like this:
- If request `https://cdn-name.com/_next/data/${build_id}/path/to/file.json` fails…
- Try using `https://current-domain.com/_next/data/${build_id}/path/to/file.json`, file should be found after the first request

A better solution should be:
- Next.js exposes a new function where you can handle files content built on runtime, so developers should upload these files on the CDN/bucket
- After uploading these files on the CDN, Next.js will find the file directly on the `https://cdn-name.com/_next/data/${build_id}/path/to/file.json` request

## System information

- OS: macOS
- Version of Next.js: 9.5.2
- Version of Node.js: 10.21.0

## Additional context

Similar issue: #16090 and #12046 

The same issue is probably effecting incremental static regeneration too

**Comment by ramiel** [comment]
Any news on this? Static generation is completely broken because of this

**Comment by vkannan-quotient** [comment]
Yeah, this is something that everyone wants. 
Or anyone help me out this issue with any working solution for assetPrefix ?

**Comment by TheBit** [comment]
@ijjk hi, may your fix https://github.com/vercel/next.js/pull/39236 help to resolve this issue as well?

**Comment by daomul** [comment]
您好，您的邮件已收到，尽快给您回复。
